src/lib.rs

```diff
@@ -1049,6 +1051,34 @@ pub enum FileMode {
     Commit,
 }
 
+impl From<u32> for FileMode {
+    fn from(mode: u32) -> Self {
+        match mode {
+            raw::GIT_FILEMODE_UNREADABLE => FileMode::Unreadable,
+            raw::GIT_FILEMODE_TREE => FileMode::Tree,
+            raw::GIT_FILEMODE_BLOB => FileMode::Blob,
+            raw::GIT_FILEMODE_BLOB_EXECUTABLE => FileMode::BlobExecutable,
+            raw::GIT_FILEMODE_LINK => FileMode::Link,
+            raw::GIT_FILEMODE_COMMIT => FileMode::Commit,
+            mode => panic!("unknown file mode: {}", mode),
```

Comment:
Since this is a fallible conversion I think it would either be best to avoid the `From` trait implementation or move this conversion to a helper function perhaps?

Reply:
Is this way ok? The `From<>` `trait` is removed
```rust
impl FileMode {
    fn from(mode: u32) -> Self {
        match mode {
            raw::GIT_FILEMODE_UNREADABLE => FileMode::Unreadable,
            raw::GIT_FILEMODE_TREE => FileMode::Tree,
            raw::GIT_FILEMODE_BLOB => FileMode::Blob,
            raw::GIT_FILEMODE_BLOB_EXECUTABLE => FileMode::BlobExecutable,
            raw::GIT_FILEMODE_LINK => FileMode::Link,
            raw::GIT_FILEMODE_COMMIT => FileMode::Commit,
            mode => panic!("unknown file mode: {}", mode),
        }
    }
}
```

Reply:
Yeah so long as this is private this should be ok 👍